content/docs/guides/ha_scale/perf-scale.md

```diff
@@ -0,0 +1,101 @@
+type: docs
+---
+
+Open Service Mesh aims to create a high-performance service mesh while providing all the essential service mesh features. This document describes the test process and results we performance to evaluate the performance and scalability of OSM.
+
+## Testing environment
+
+This test is performed on a Kubernetes cluster provisioned and managed by [Microsoft Azure Kubernetes Service](https://azure.microsoft.com/en-us/services/kubernetes-service/). The cluster has 100 Standard_DS2_v2 nodes (each node has 2 vCPUs and 7GB memory). It is ensured that there is only one OSM controller pod running with default resource quota during the test. OSM is configured in non-permissive traffic mode.
+
+## Mesh topology
+
+The testing mesh contains two types of service. One is a load generator (modified from [wrk2](https://github.com/giltene/wrk2)), which can send requests at specified rate, load balanced among all other services in the mesh. The load generator serves as the client side of a request. Throughout our test, there will be at most one load generator service. The other type of service is a simple echo application that replies with whatever it receives from an inbound request. It serves as the server side of a request. We can deploy multiple copies of the echo application in the mesh. In short, the mesh topology is a typical 1-to-N service architecture.
+
+<p align="center">
+  <img src="/docs/images/perf/mesh-topo.png" width="650"/>
+</p>
+
+Note that since both types of service are in the mesh and permissive mode is disabled, thus each request will be authenticated via mTLS. It means that requests go through the Envoy sidecar in both the client and the server.
+
+## Metrics
+
+The test result includes a lot of metrics collected. A clear definition of each metric can help avoid ambiguity. Below are the metric definitions we will use throughout this documentation:
+
+* Control plane
+    * Peak time CPU/Memory: the maximum CPU/memory usage during the testing process.
+    * Stable time CPU/Memory usage: the average CPU/memory usage when the control plane is off the peak time and becomes stable.
+* Data plane sidecar
+    * Peak time CPU/Memory: the maximum CPU/memory usage among all sidecar containers during the testing process.
+    * Stable time CPU/Memory usage: the average CPU/memory usage of all sidecar containers after they become stable.
+* Request latency: request roundtrip time. P50, P90, P99 and P999 of request latency are measured.
+
+The reason that peak time and stable time are measured individually is because of how the services are deployed. All the echo services will be deployed and added to the mesh at the same time. Therefore, the control plane receives a load spike after the deployment starts. All the Envoy sidecars are connecting to the control plane simutaneously. After that, the control plane becomes much more idle, since there is no more change to the mesh topology.
```

Comment:
Ok so this makes sense, but I think the above is misleading. If we want to delineate metrics during periods of high churn, lets call that out in the definition, and not as a note here :)


The data plane should largely be unaffected by this churn, so let's just drop peak/stable terminology for that altogether.

The control plane we an call that out, but I don't think it's useful to plot them on the same graph. Let's separate that out is a completely different scenario (there may be differing opinions on this, let's get others' feedback)

Reply:
Adding more context to the graph separation:

I would think of that as average and max (which is a valid thing to do **if** there are multiple backends we are measuring (true for sidecar, not currently true for osm-controller, although a future scenario may change that). So that's what I expect when I see that graph, and why I think we should separate it.